### PR TITLE
Add file extension info to the system prompt

### PR DESCRIPTION
**Pull request by nilszeilon**
Added file extension info to the system prompt, sometimes it would give me code in another language than the intended if that information was missing.

### CODE REVIEW

**Review by yacineMTB — CHANGES_REQUESTED** [review]
Thanks for this, this is a great idea.
Do you think we could instead add a function that constructs the system prompt? So that we can change the function in config to programatically construct system prompts based on runtime context

**Comment by nilszeilon** [comment]
I agree, that is cleaner. Now your system prompt can be passed through the function 'create_prompt' which replaces $file_ext variable in the system prompt with the actual file extension. One can easily add more custom variables that will be calculated from the current context.

**Comment by adharm** [comment]
Would it would be more extensible/easier to update by putting the constructor in the config? Then dingllm.lua wouldn't have to change every time you wanted to add more context. So something like:
```
return {
...
    local helpful_prompt = 'You are a helpful assistant. What I have sent are my notes so far. You are very curt, yet helpful.'

    local function construct_replace_prompt()
      local base_replace_prompt =
        'You should replace the code that you are sent, only following the comments. Do not talk at all. Only output valid code. 
         Do not provide any backticks that surround the code. Never ever output backticks like this ```. Any comment that is 
         asking you for something should be removed after you satisfy them. Other comments should left alone. Do not output 
          backticks.'
      -- append file type to prompt
      local current_file_type = vim.bo.filetype
      local replace_prompt = base_replace_prompt .. ' Note that this is a ' .. current_file_type .. ' file.'
      -- add other prompt additions as needed
      return replace_prompt
    end

    local function groq_replace()
      dingllm.invoke_llm_and_stream_into_editor({
        url = 'https://api.groq.com/openai/v1/chat/completions',
        model = 'llama-3.1-70b-versatile',
        api_key_name = 'GROQ_API_KEY',
        system_prompt = construct_replace_prompt(),
        replace = true,
      }, dingllm.make_openai_spec_curl_args, dingllm.handle_openai_spec_data)
    end

    local function groq_help()
      dingllm.invoke_llm_and_stream_into_editor({
        url = 'https://api.groq.com/openai/v1/chat/completions',
        model = 'llama-3.1-70b-versatile',
        api_key_name = 'GROQ_API_KEY',
        system_prompt = helpful_prompt,
        replace = false,
      }, dingllm.make_openai_spec_curl_args, dingllm.handle_openai_spec_data)
    end
 ...
 }
```

File: lua/dingllm.lua [hunk 1]
```diff
@@ -5,6 +5,21 @@ local function get_api_key(name)
...
+end
+
+function M.create_prompt(prompt)
+  local ext = get_file_extension()
+  if ext then
+    prompt = prompt:gsub("$file_ext", ext)
+  end
+  return prompt
+end
+
```
Comment (yacineMTB) [inline lua/dingllm.lua:22]: Sorry for the back and forth and also lag. But I meant actually make system prompt a variable that can be either a function or string. If it's a function, take whatever it returns. If it's a string, use as is. If that makes sense!

**Review by yacineMTB — CHANGES_REQUESTED** [review]

**Pull request closed without merging**